### PR TITLE
Text Converter: Add padding to binary output

### PR DESCRIPTION
**Pull request by pjhampton**
## Description of new Instant Answer, or changes

Adds padding to binary output.

screen shot 2017-07-31 at 6 35 06 pm

## People to notify

@moollaza 

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/text_converter

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by moollaza** [comment]
@pjhampton Thanks! This LGTM :+1:

File: share/goodie/text_converter/text_converter.js [hunk 1]
```diff
@@ -78,17 +78,20 @@ DDH.text_converter = DDH.text_converter || {};
...
-            var binary = "";
+        zeroPad: function( number ) {
+            return "00000000".slice(String(number).length) + number
+        },
 
-            for (var i=0; i < text.length; i++) {
-                binary += text[i].charCodeAt(0).toString(2) + " ";
-            }
-            return binary
+        toBinary: function( text, octets ) {
```
Comment (moollaza) [inline share/goodie/text_converter/text_converter.js]: it doesn't look like we ever pass in an `octets` value to the `toBinary` function?

File: share/goodie/text_converter/text_converter.js [hunk 2]
```diff
@@ -78,17 +78,20 @@ DDH.text_converter = DDH.text_converter || {};
...
+            return text.replace(/[\s\S]/g, function(str) {
+                text = TextConverter.zeroPad(str.charCodeAt().toString(2));
+                return !1 == octets ? text : text + " "
```
Comment (moollaza) [inline share/goodie/text_converter/text_converter.js]: the `!1 ==` is a little odd IMO. I would `return octets === false ? .....`

**Comment by bsstoner** [comment]
This was released to production 🚀

**Pull request merged**